kong/core/router.lua

```diff
@@ -637,12 +637,25 @@ function _M.new(apis)
     end
 
 
-    if not uri_root and api_t.strip_uri_regex then
-      local err
-      new_uri, err = re_sub(uri, api_t.strip_uri_regex, "/$1", "ajo")
-      if not new_uri then
-        log(ERR, "could not strip URI: ", err)
-        return
+    if uri_root then
+      new_uri = uri
+
+    else
+      local s = find(uri, "?", 2, true)
+      if s then
+        uri = sub(uri, 1, s - 1)
```

Comment:
As a side note, it may be worth it to introduce unit tests that ensure the cache is hit when appropriate in the future. Part of a future improvement we can do to ensure the stability of such a critical component :)